scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -257,6 +258,17 @@ object Execution {
   def withConfig[T](ex: Execution[T])(c: Config => Config): Execution[T] =
     TransformedConfig(ex, c)
 
+  def withCached[T](path: String)(fn: CachedFile => Execution[T]): Execution[T] = {
```

Comment:
can we add a comment that you cannot return `CachedFile` in `T`. You must use `CachedFile` inside something inside the `Execution[T]` (like a map/reduce function).